extensions-contrib/opentelemetry-extensions/src/main/java/org/apache/druid/data/input/opentelemetry/protobuf/traces/OpenTelemetryTracesProtobufInputFormat.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.druid.data.input.opentelemetry.protobuf.traces;
+
+import com.fasterxml.jackson.annotation.JsonProperty;
+import org.apache.druid.data.input.InputEntity;
+import org.apache.druid.data.input.InputEntityReader;
+import org.apache.druid.data.input.InputRowSchema;
+import org.apache.druid.data.input.opentelemetry.protobuf.OpenTelemetryInputFormat;
+
+import java.io.File;
+import java.util.Objects;
+
+public class OpenTelemetryTracesProtobufInputFormat extends OpenTelemetryInputFormat
+{
+
+  private final OpenTelemetryTracesProtobufConfiguration config;
+
+  @JsonProperty
+  public OpenTelemetryTracesProtobufConfiguration getConfig()
+  {
+    return config;
+  }
+
+  public OpenTelemetryTracesProtobufInputFormat(
+      @JsonProperty("config") OpenTelemetryTracesProtobufConfiguration config
```

Comment:
I don't mind changing it, but any specific reason why we shouldn't encapsulate? 

Given there are 10 configurable fields, encapsulating them looks cleaner than following the existing pattern.

Reply:
the InputFormat constructor is also the configuration for that format, that's how all formats work today (see https://druid.apache.org/docs/latest/ingestion/data-formats.html#input-format) it would be odd for this one to not follow the existing conventions.

Adding a level of nesting also doesn't seem to add much here, you still have 10 fields on the config object, and it makes the configuration more error-prone to write since you now have to remember to nest everything under `config` unlike all other input format.